docs/getting-started-guides/mesos-docker.md

```diff
@@ -0,0 +1,201 @@
+
+### Cluster Goals
+
+- kubernetes development
+- pod/service development
+- demoing
+- fast deployment
+- minimal hardware requirements
+- minimal configuration
+- entry point for exploration
+- simplified networking
+- fast end-to-end tests
+- local deployment
+
+Non-Goals:
+- high availability
+- fault tolerance
+- remote deployment
+- production usage
+- monitoring
+- long running
+- state persistence across restarts
+
+### Cluster Topology
+
+The cluster consists of several docker containers linked together by docker-managed hostnames:
+
+| Component                     | Hostname                    | Description                                                                             |
+|-------------------------------|-----------------------------|-----------------------------------------------------------------------------------------|
+| docker-grand-ambassador       |                             | Proxy to allow circular hostname linking in docker                                      |
+| etcd                          | etcd                        | Key/Value store used by Mesos                                                           |
+| Mesos Master                  | mesosmaster1                | REST endpoint for interacting with Mesos                                                |
+| Mesos Slave (x2)              | mesosslave1<br/>mesosslave2 | Mesos agents that offer resources and run framework executors (e.g. Kubernetes Kublets) |
+| Kubernetes API Server         | apiserver                   | REST endpoint for interacting with Kubernetes                                           |
+| Kubernetes Controller Manager | controller                  |                                                                                         |
+| Kubernetes Scheduler          | scheduler                   | Schedules container deployment by accepting Mesos offers                                |
+
+### Prerequisites
+
+- [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+- [Docker CLI](https://docs.docker.com/)
+- [Docker Engine](https://docs.docker.com/)<br/>
+    On Mac, use [Boot2Docker](http://boot2docker.io/) or [Docker Machine](https://docs.docker.com/machine/install-machine/)
+    to run Docker Engine in a linux VM.
+- [Optional] [Virtual Box](https://www.virtualbox.org/wiki/Downloads)<br/>
+    Required by Boot2Docker and Docker Machine
+- [Optional] [etcd](https://github.com/coreos/etcd)<br/>
+    Only used locally by integration tests, when building without `KUBE_RELEASE_RUN_TESTS=N`.
+
+Note: On Mac, it's possible to install all the above via [Homebrew](http://brew.sh/).<br/>
+Follow any printed instructions after each step to make sure each is configured correctly.
+
+```
+brew install git
+brew install caskroom/cask/brew-cask
+brew cask install virtualbox
+brew install docker
+brew install boot2docker
+boot2docker init
+boot2docker up
+brew install etcd
+```
+
+***TODO***: apt & yum instructions
```

Comment:
That's a valid concern. boot2docker is nearly outdated already, with the new docker machine. I just wanted to document what I use that I know works. In the future, I want to have unit & integration tests run in a docker container so that go and etcd aren't required locally. But git and docker are always going to be required...